IdentityCore/src/cache/MSIDTokenCacheKey.m

```diff
@@ -39,4 +47,106 @@ - (id)initWithAccount:(NSString *)account
     return self;
 }
 
+//We should not put nil keys in the keychain. The method substitutes nil with a special GUID:
++ (NSString *)getAttributeName:(NSString *)original
+{
+    return ([NSString msidIsStringNilOrBlank:original]) ? s_nilKey : [original msidBase64UrlEncode];
+}
+
++ (NSString *)serviceWithAuthority:(NSURL *)authority
+                          resource:(NSString *)resource
+                          clientId:(NSString *)clientId
+{
+    
+    return [NSString stringWithFormat:@"%@|%@|%@|%@",
+            s_adalLibraryString,
+            authority.absoluteString.msidBase64UrlEncode,
+            [self.class getAttributeName:resource.msidBase64UrlEncode],
+            clientId.msidBase64UrlEncode];
+}
+
++ (NSString *)accountWithUserIdentifier:(NSString *)userId
+                            environment:(NSString *)environment
+{
+    return userId? [NSString stringWithFormat:@"%u$%@@%@", s_msalV1, userId, environment]: nil;
```

Comment:
I think we should take out s_msalV1?

Reply:
yes, when we finally have a conclusion, i plan to remove it.